### PR TITLE
Update CI to use new nebari cli entrypoint

### PR DESCRIPTION
**Pull request by iameskild**
Fixes | Closes | Resolves #

> Please remove anything marked as optional that you don't need to fill in.
> Choose one of the keywords preceding to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666).
> If there is no issue, remove the line. Remove this note after reading.

## Changes introduced in this PR:

- Update CI to use the new Nebari, typer based CLI. 

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [x] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [x] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

File: .github/workflows/kubernetes_test.yaml [hunk 1]
```diff
@@ -73,32 +73,32 @@ jobs:
...
         run: |
-          sudo echo "172.18.1.100  github-actions.qhub.dev" | sudo tee -a /etc/hosts
-      - name: Initialize QHub Cloud
+          sudo echo "172.18.1.100  github-actions.nebari.dev" | sudo tee -a /etc/hosts
+      - name: Initialize Nebari Cloud
         run: |
           mkdir -p local-deployment
           cd local-deployment
-          qhub init local --project=thisisatest --domain github-actions.qhub.dev --auth-provider=password
+          nebari init local --project=thisisatest --domain github-actions.nebari.dev --auth-provider=password
```
Comment (viniciusdc) [inline .github/workflows/kubernetes_test.yaml:81]: @iameskild How is this working? Did we copy the `qhub-config.yaml` or rename it somewhere?

Comment (viniciusdc) [reply .github/workflows/kubernetes_test.yaml:81]: https://github.com/Quansight/qhub/blob/089da20126f28cdd3a54b9f27c4efed09e3d9b75/qhub/cli/_init.py#L80-L86

File: .github/workflows/kubernetes_test.yaml [hunk 2]
```diff
@@ -73,32 +73,32 @@ jobs:
...
           # Need smaller profiles on Local Kind
-          sed -i -E 's/(cpu_guarantee):\s+[0-9\.]+/\1: 0.25/g' "qhub-config.yaml"
-          sed -i -E 's/(mem_guarantee):\s+[A-Za-z0-9\.]+/\1: 0.25G/g' "qhub-config.yaml"
+          sed -i -E 's/(cpu_guarantee):\s+[0-9\.]+/\1: 0.25/g' "nebari-config.yaml"
+          sed -i -E 's/(mem_guarantee):\s+[A-Za-z0-9\.]+/\1: 0.25G/g' "nebari-config.yaml"
 
-          cat qhub-config.yaml
-      - name: Deploy QHub Cloud
+          cat nebari-config.yaml
+      - name: Deploy Nebari Cloud
```
Comment (viniciusdc) [inline .github/workflows/kubernetes_test.yaml:88]:
```suggestion
      - name: Deploy Nebari
```

File: .github/workflows/kubernetes_test.yaml (hunk 1 above)
Comment (iameskild) [reply .github/workflows/kubernetes_test.yaml:81]: I did this for the time being:
https://github.com/Quansight/qhub/pull/1480/files#diff-322d8009fe7ede1efc69ad7e72d8985e15edb595e0c8588fa5a66936bc053e0cR84

File: .github/workflows/kubernetes_test.yaml [hunk 3]
```diff
@@ -113,8 +113,8 @@ jobs:
           sudo apt-get -y update
           sudo apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
 
-      - name: Get qhub-config.yaml full path
-        run: echo "QHUB_CONFIG_PATH=`realpath ./local-deployment/qhub-config.yaml`" >> "$GITHUB_ENV"
+      - name: Get nebari-config.yaml full path
+        run: echo "NEBARI_CONFIG_PATH=`realpath ./local-deployment/nebari-config.yaml`" >> "$GITHUB_ENV"
```
Comment (viniciusdc) [inline .github/workflows/kubernetes_test.yaml:117]: Did we change this env as well?

Comment (viniciusdc) [reply .github/workflows/kubernetes_test.yaml:117]: oh, nvm just saw the rest

File: .github/workflows/kubernetes_test.yaml (hunk 1 above)
Comment (viniciusdc) [reply .github/workflows/kubernetes_test.yaml:81]: nvm, just red it bellow

File: qhub/cli/main.py [hunk 4]
```diff
@@ -319,13 +326,18 @@ def destroy(
...
 ):
     """
     Destroy the Nebari cluster from your [purple]nebari-config.yaml[/purple] file.
     """
-    delete = typer.confirm("Are you sure you want to destroy it?")
-    if not delete:
-        raise typer.Abort()
+    if not disable_prompt:
+        if typer.confirm("Are you sure you want to destroy your Nebari cluster?"):
+            raise typer.Abort()
```
Comment (viniciusdc) [inline qhub/cli/main.py:340]: Good catch, we need this for running this in the Action

**Review by viniciusdc — APPROVED** [review]
LGTM!! Thanks @iameskild !

File: .github/workflows/kubernetes_test.yaml [hunk 5]
```diff
@@ -155,15 +155,15 @@ jobs:
...
                                 --stop-server \
                                 --notebook tests_deployment/assets/notebook/simple.ipynb \
 
       ### CLEANUP AFTER TESTS
-      - name: Cleanup qhub deployment
+      - name: Cleanup nebari deployment
         run: |
           cd local-deployment
-          qhub destroy --config qhub-config.yaml
+          nebari destroy --config nebari-config.yaml --disable-prompt
```
Comment (trallard) [inline .github/workflows/kubernetes_test.yaml:169]:
```suggestion
		  # need to bypass user confirmation prompt
          nebari destroy --config nebari-config.yaml --disable-prompt
```

added a comment for clarity

File: qhub/cli/main.py [hunk 6]
```diff
@@ -319,13 +326,18 @@ def destroy(
         "--disable-render",
         help="Disable auto-rendering before destroy",
     ),
+    disable_prompt: bool = typer.Option(
+        False,
+        "--disable-prompt",
+        help="Destroy entire Nebari cluster without confirmation request",
```
Comment (trallard) [inline qhub/cli/main.py:332]:
```suggestion
        help="Destroy entire Nebari cluster without confirmation request. Suggested for CI use.",
```

**Review by trallard — COMMENTED** [review]
LGTM - there are loads of `TODO` for when the CLI is out of beta. Would like to discuss when this will be

**Comment by iameskild** [comment]
Thanks @trallard and @viniciusdc for the review! There are 4 TODOs all related to how we name the config file. These will all be torn out when we fully switch over, in the meantime, this allows for us to use both CLIs and have them behave in expected ways.

**Pull request merged**